src/server/routes/editor.js

```diff
@@ -250,47 +251,93 @@ router.get('/:id', (req, res, next) => {
 	);
 });
 
-router.get('/:id/revisions', (req, res, next) => {
+async function getOrderedRevisionForEditorPage(from, size, req) {
+	const {Note, Revision} = req.app.locals.orm;
+
+	const revisions = await new Revision()
+		.query('where', 'author_id', '=', parseInt(req.params.id, 10))
+		.orderBy('created_at', 'DESC')
+		.fetchPage({
+			limit: size,
+			offset: from,
+			withRelated: [
+				'author'
+			]
+		});
+	const revisionsJSON = revisions.toJSON();
+
+	const formattedRevisions = await Promise.all(
+		revisionsJSON.map(async rev => {
+			const note = await new Note()
+				.query('where', 'revision_id', '=', parseInt(rev.id, 10))
+				.fetch({require: false});
+			const noteContent = note ? note.toJSON().content : ' ';
+
+			const {author: editor, id: revisionId, ...otherProps} = rev;
+			return {editor, entities: [], noteContent, revisionId, ...otherProps};
+		}));
+
+
+	const orderedRevisions = await utilis.getAssociatedEntityRevisions(formattedRevisions, req.app.locals.orm);
+
+	return orderedRevisions;
+}
+
+router.get('/:id/revisions', async (req, res, next) => {
 	const {Editor, TitleUnlock} = req.app.locals.orm;
-	new Editor({id: parseInt(req.params.id, 10)})
-		.fetch({
-			require: true,
-			withRelated: {
-				revisions(query) {
-					query.orderBy('id');
-				}
-			}
-		})
-		.then((editor) => getEditorTitleJSON(editor.toJSON(), TitleUnlock))
-		.then((editorJSON) => {
-			const props = generateProps(req, res, {
-				editor: editorJSON,
-				tabActive: 1
-			});
-			const markup = ReactDOMServer.renderToString(
-				<Layout {...propHelpers.extractLayoutProps(props)}>
-					<EditorContainer
-						{...propHelpers.extractEditorProps(props)}
-					>
-						<RevisionsTab
-							editor={props.editor}
-						/>
-					</EditorContainer>
-				</Layout>
-			);
-			res.send(target({
-				markup,
-				page: 'revisions',
-				props: escapeProps(props),
-				script: '/js/editor/editor.js'
-			}));
-		})
-		.catch(Editor.NotFoundError, () => {
-			throw new error.NotFoundError('Editor not found', req);
-		})
-		.catch(next);
+
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	const orderedRevisions = await getOrderedRevisionForEditorPage(from, size, req);
+
+	const editor = await new Editor({id: req.params.id}).fetch();
+	const editorJSON = await getEditorTitleJSON(editor.toJSON(), TitleUnlock);
+
+	const props = generateProps(req, res, {
+		editor: editorJSON,
```

Comment:
This editor prop seems to be unused now (editor name was displayed on each revision previously).

If so, it could either be deleted entirely (no need to `getEditorTitleJSON` is that case), or used for the RevisionsPage title ("Revision History for {editorName}"). I'll let you have a look at what works out best.

Reply:
editor props is used actually. Not in the table but we need editor.id and editor.name for url and heading.(Sprinter Monkey). You were right though, we don't need to fetch authors (withRelated) while getting revisions. :)